pty/pty_other.go

```diff
@@ -66,3 +77,21 @@ func (p *otherPty) Close() error {
 	}
 	return nil
 }
+
+func (p *otherPtyWithProcess) Wait() error {
+	<-p.cmdDone
+	return p.cmdErr
+}
+
+func (p *otherPtyWithProcess) Kill() error {
+	return p.cmd.Process.Kill()
+}
+
+func (p *otherPtyWithProcess) waitInternal() {
```

Comment:
I'm not entirely sure I understand why we need to abstract this. It seems both implementations use the `process` and wait. I feel like we could switch based on OS in the `Start` func instead, and still return the underlying `*os.Process`.

Reply:
It's not safe for multiple goroutines to Wait() on an `os.Process`.  This creates a race where one of them can get a SyscallError complaining that the child process doesn't exist.

We didn't notice this before because we were throwing away the errors in a TTY, but we do actually need them to return the correct exit code.

Reply:
Ahh I see. What do you think about returning an abstract Process in that case instead of with the TTY? Seems like they are separate concerns

Reply:
Yeah, I can separate them out.  The underlying impl will need a reference to the pty in the Linux case, so that we can call `runtime.KeepAlive`, but that will be hidden from the caller.